R/functions.R

```diff
@@ -1258,73 +1260,113 @@ test_gene_enrichment_bulk_EGSEA <- function(.data,
 		spread(!!.sample,!!.abundance) %>%
 		as_matrix(rownames = !!.entrez) %>%
 		edgeR::DGEList(counts = .)
+    
+	# Specify gene sets to include 
+	if ("all" %in% gene_collections) {
+	    msigdb.gsets <- "all"
+	    kegg.exclude <- c()
+	} else {
+	    msig <- c("h", "c1", "c2", "c3", "c4", "c5", "c6", "c7")
+	    kegg <- c("Disease", "Metabolism", "Signaling")
```

Comment:
Amazing. I though one thing it would really simplify things for users is having the argument (the way egsea treats arguments is really badly designed in my opinion, and exposes complexities that the users should not care about)

The cleanest thing would be
gene_collection = c("h", "c1", "c2", "c3", "c4", "c5", "c6", "c7", "kegg_Disease", "kegg_Metabolism", "kegg_Signaling")

internally we can do
`gene_collection %>% str_replace("kegg_", "")`
and proceed to parsing

Reply:
Do "Disease", "Metabolism", "Signaling" need to be sentence case for egsea/msigdb to work?

Reply:
>Do "Disease", "Metabolism", "Signaling" need to be sentence case for egsea/msigdb to work?

Apparently not, I just tested it and all lowercase works fine.

>gene_collection = c("h", "c1", "c2", "c3", "c4", "c5", "c6", "c7", "kegg_Disease", "kegg_Metabolism", "kegg_Signaling")

Maybe I'll change to "kegg_disease", "kegg_metabolism", "kegg_signaling" then to try to reduce any confusion.

Reply:
The reason why I like `gene_collection = c("h", "c1", "c2", "c3", "c4", "c5", "c6", "c7", "kegg_Disease", "kegg_Metabolism", "kegg_Signaling")` is because a lot can be understood without documntation

Reply:
Yes agree, I've pushed that change now.

Reply:
Sorry all lower case if can be accepted by msigdb

gene_collection = c("h", "c1", "c2", "c3", "c4", "c5", "c6", "c7", "kegg_disease", "kegg_metabolism", "kegg_signaling")

Reply:
Yep all lowercase is what I did.